README.md

```diff
@@ -218,6 +219,58 @@ class SomeConfiguration: Configuration {
 
 > There is a quick way to get the correct public key for a certain domain. Go to [SSL Server Test](https://www.ssllabs.com/ssltest/) by SSL Labs in order to perform an analysis of the SSL configuration of any web server. In the inputfield you enter the domain in order to get the process started. On the next page click the first IP address that appears, and on the page after, you'll notice the `Pin SHA256` field. The value is the public key string we need.
 
+### Network Reachability
+
+NetworkReachability is a wrapper, providing a reliable measure of whether Internet connectivity is available.
+
+#### Listener
+
+To get started using NetworkReachability, simply instantiate an instance and assign a closure to be invoked when NetworkReachability detects that you are connected to the Internet, when disconnected, or in both cases as follows:
+
+```swift
+let reachability = NetworkReachability()
+reachability.listener = { status in
```

Comment:
I wouldn't make an extra class for this, internally okay. But not exposed to the user.

ex.
```swift
service.networkStatusChange = { status in
}
```

This way the `Service` instance remains the main instance and we can the base url configuration when we really want to start pinging for the correct source. You can set different [Connectivity URLs](connectivity.connectivityURLs), this could be the base url or something similar that is available on backend side.